### PR TITLE
TILA-622 | Reservation and Recurring reservation api adds

### PR DESCRIPTION
**Pull request by nicobav**

### CODE REVIEW

File: api/reservations_api.py [hunk 1]
```diff
@@ -85,6 +160,22 @@ class Meta:
             },
         }
 
+    def get_begin_weekday(self, instance):
+        return instance.begin.weekday()
+
+    def get_application_event_name(self, instance):
```
Comment (aprenfors) [inline api/reservations_api.py:166]: Not related to this pull request, but i think this whole recurring reservations seems quite unnecessary, could just link reservations straight to application events imo. 

But let's see about refactoring that at some time.

**Review by aprenfors — APPROVED** [review]

File: api/reservations_api.py (hunk 1 above)
Comment (nicobav) [reply api/reservations_api.py:166]: yes, that is true, this `RecurringReservation`  might be unnecessary boilerplate model in the middle.

Comment (aprenfors) [reply api/reservations_api.py:166]: Yeah, it's from way back from initial models, when we didn't even have any actual recurring reservations and stuff.

Comment (nicobav) [reply api/reservations_api.py:166]: we could think of the v2 api at near future and drop this then.

**Pull request merged**